src/metrics/misc.ts

```diff
@@ -17,3 +26,163 @@ export function countGamesInDataset(datasetPath: string): number {
+        maxGames = this.playerGameStats[player];
+        playerMostGames = player;
+      }
+    }
+    this.mostGamesPlayed = maxGames;
+    this.playerMostGames = playerMostGames;
+
+    return {
+      averagePlayerRating: this.averagePlayerRating,
+      averageRatingDiff: this.averageRatingDiff,
+      mostGamesPlayed: maxGames,
+      playerMostGames,
+    };
+  }
+
+  // Reset the maps used to track metrics
+  clear(): void {
+    this.numberGamesAnalyzed = 0;
+    this.averagePlayerRating = 0;
+    this.averageRatingDiff = 0;
+    this.largestRatingDiff = 0;
+    this.largestRatingDiffGame = '';
+    this.mostGamesPlayed = 0;
+    this.playerMostGames = '';
+    this.gameTypeStats = {
+      numberBulletGames: 0,
+      numberBlitzGames: 0,
+      numberRapidGames: 0,
+      numberClassicalGames: 0,
+    };
+    this.gameTimeControlStats = {};
+
+    this.totalPlayerRating = 0;
+    this.totalPlayerRatingDiff = 0;
+    this.playerGameStats = {};
+
+  }
+
+  processGame(
+    game: { move: PrettyMove; board: Piece[] }[],
+    metadata?: string[]
+  ) {
+
+    // Update the gameTimeControlStats based on the time control of the game
+    const timeControl = metadata?.find((data) => data.startsWith('[TimeControl'));
+    if (timeControl) {
+      const timeControlQuantity = timeControl.split(' ')[1];
+      if (this.gameTimeControlStats[timeControlQuantity]) {
+        this.gameTimeControlStats[timeControlQuantity]++;
+      } else {
+        this.gameTimeControlStats[timeControlQuantity] = 1;
+      }
+    }
+    
+    // Identify the time control type from the metadata
+    const gameType = metadata?.find((data) => data.startsWith('[Event'));
+    if (gameType) {
+      if (gameType.includes('Bullet')) {
+        this.gameTypeStats.numberBulletGames++;
+      } else if (gameType.includes('Blitz')) {
+        this.gameTypeStats.numberBlitzGames++;
+      } else if (gameType.includes('Rapid')) {
+        this.gameTypeStats.numberRapidGames++;
+      } else if (gameType.includes('Classical')) {
+        this.gameTypeStats.numberClassicalGames++;
+      }
+    }
+
+    // Calculate average player rating (first, find the totalPlayerRating)
+    const whiteRating = parseInt(metadata?.find((data) => data.startsWith('[WhiteElo'))?.replace(/"/g, '').split(' ')[1]);
+    const blackRating = parseInt(metadata?.find((data) => data.startsWith('[BlackElo'))?.replace(/"/g, '').split(' ')[1]);
+    this.totalPlayerRating += whiteRating + blackRating;
+
+    // Calculate the player rating diffs
+    const ratingDiff = Math.abs(whiteRating - blackRating);
+    this.totalPlayerRatingDiff += ratingDiff;
+
+    // Check if this game has the largest rating differential
+    if (ratingDiff > this.largestRatingDiff) {
+      this.largestRatingDiff = ratingDiff;
+      this.largestRatingDiffGame = metadata?.find((data) => data.startsWith('[Site')) || '';
+    }
+
+    // Calculate the player with the most games played
```

Comment:
some comments here don't make sense to me

Reply:
added better comments